streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -232,15 +267,24 @@ public StreamsConfig(Map<?, ?> props) {
         props.put(StreamsConfig.REPLICATION_FACTOR_CONFIG, getInt(REPLICATION_FACTOR_CONFIG));
         props.put(StreamsConfig.NUM_STANDBY_REPLICAS_CONFIG, getInt(NUM_STANDBY_REPLICAS_CONFIG));
         props.put(ConsumerConfig.PARTITION_ASSIGNMENT_STRATEGY_CONFIG, StreamPartitionAssignor.class.getName());
-
         if (!getString(ZOOKEEPER_CONNECT_CONFIG).equals(""))
             props.put(StreamsConfig.ZOOKEEPER_CONNECT_CONFIG, getString(ZOOKEEPER_CONNECT_CONFIG));
 
         return props;
     }
 
-    public Map<String, Object> getRestoreConsumerConfigs(String clientId) {
-        Map<String, Object> props = getBaseConsumerConfigs();
+    public Map<String, Object> getRestoreConsumerConfigs(String clientId) throws ConfigException {
+        // first disable auto commit ignoring any user overridden values,
+        // this is necessary for streams commit semantics
+        Map<String, Object> originals = this.originals();
+
+        if (originals.containsKey(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG)) {
+            throw new ConfigException("Unexpected user-specified consumer config " + ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG
+                    + ", as the streams client will always turn off auto committing.");
+        }
+
+        // generate consumer configs from original properties and overridden maps
+        Map<String, Object> props = clientProps(ConsumerConfig.configNames(), originals, CONSUMER_DEFAULT_OVERRIDES);
```

Comment:
Since the disabling of auto commit is specific to Streams, I'd prefer to keep them here.